Modules/System/Azure Blob Services API/src/Helper/ABSOptionalParameters.Codeunit.al

```diff
@@ -226,6 +226,66 @@ codeunit 9047 "ABS Optional Parameters"
         SetRequestHeader('x-ms-requires-sync', ValueText);
     end;
 
+    /// <summary>
+    /// Sets the value for 'x-ms-lease-action' HttpHeader for a request.
+    /// </summary>
+    /// <param name="Value">Enum "ABS Lease Action" value specifying the HttpHeader value</param>    
+    internal procedure LeaseAction("Value": Enum "ABS Lease Action")
+    begin
+        SetRequestHeader('x-ms-lease-action', Format("Value"));
+    end;
+
+    /// <summary>
+    /// Sets the value for 'x-ms-lease-break-period' HttpHeader for a request.
+    /// </summary>
+    /// <param name="Value">Integer value specifying the HttpHeader value.</param>
+    procedure LeaseBreakPeriod("Value": Integer)
+    var
+        LeaseAction: Enum "ABS Lease Action";
+    begin
+        LeaseAction := GetLeaseActionFromRequestHeader();
+        if not (LeaseAction in [LeaseAction::break]) then
+            Error(HeaderCanOnlyBeSetOnConditionErr, 'x-ms-lease-break-period', 'x-ms-lease-action', 'break');
+        SetRequestHeader('x-ms-lease-break-period', Format("Value"));
+    end;
+
+    /// <summary>
+    /// Sets the value for 'x-ms-lease-duration' HttpHeader for a request.
+    /// </summary>
+    /// <param name="Value">Integer value specifying the HttpHeader value.</param>
+    procedure LeaseDuration("Value": Integer)
+    var
+        LeaseAction: Enum "ABS Lease Action";
+    begin
+        LeaseAction := GetLeaseActionFromRequestHeader();
+        if not (LeaseAction in [LeaseAction::acquire]) then
+            Error(HeaderCanOnlyBeSetOnConditionErr, 'x-ms-lease-duration', 'x-ms-lease-action', 'acquire');
+        SetRequestHeader('x-ms-lease-duration', Format("Value"));
+    end;
+
+    /// <summary>
+    /// Sets the value for 'x-ms-proposed-lease-id' HttpHeader for a request.
+    /// </summary>
+    /// <param name="Value">Guid value specifying the HttpHeader value.</param>
+    procedure ProposedLeaseId("Value": Guid)
```

Comment:
Suggestion: `SetProposedLeaseId`

Reply:
This isn't how this codeunit rolls, the setters are anonymous 😄